### ISSUE TITLE
cross-module class connectors ( inheritance, association, dependency)

### ISSUE THREAD

**Issue by costezki** [OP-TED/model2owl#142]
*Generalisation relation example*
`epo:ResultNotice` (in eNotice module) inherits from epo:Notice (in Core module) in the UML model. 
However, when OWL is generated, the connectors across modules are lost (i.e. are not included in the transformation output). We need to generate such ingeritance relations in the eNotice OWL module. 

The same principle applies to other connector types (generalisation, association, dependency, ). The general rule is: 
`if a class in module A has a relation to a class in module B, then that relation needs to be reflected when the module A is transformed.`

**Comment by cristianvasquez** [comment]
We are facing this problem for e-Access (https://github.com/OP-TED/ePO/pull/586). The generated artifacts don't have all relations. Is this related to the XMI exports? (related #146)

Towards fixing the issue, I saw there is a merge-xmi operation in the repository, is it used for that purpose?

**Comment by cristianvasquez** [comment]
More information:

See the XMI for one module:

https://raw.githubusercontent.com/OP-TED/ePO/develop/implementation/eSubmission/xmi_conceptual_model/eSubmission.xml

Take one of he inter-module relations in the file: refersToNotice
```xml
<ownedElement xmi:type="umldi:UMLAssociationEndLabel" xmi:id="EAID_TEL000000_3004_48e5_963A_7FE0F57DA5D8" text="+epo-acc:refersToNotice" modelElement="EAID_dstDD6F7E_CC4D_4861_963A_7FE0F57DA5D8">
    <bounds xmi:type="dc:bounds" xmi:id="EAID_DBTEL000000_3004_48e5_963A_7FE0F57DA5D8" x="295" y="417" width="98" height="14"/>
</ownedElement>
<ownedElement xmi:type="umldi:UMLAssociationEndLabel" xmi:id="EAID_TEL000000_3004_48e5_BE52_91F3F7270840" text="+epo:refersToNotice 0..1" modelElement="EAID_dstE70932_F57C_4861_BE52_91F3F7270840">
    <bounds xmi:type="dc:bounds" xmi:id="EAID_DBTEL000000_3004_48e5_BE52_91F3F7270840" x="81" y="529" width="100" height="14"/>
</ownedElement>
```
~~They refer to the modelElement id: EAID_dstDD6F7E_CC4D_4861_963A_7FE0F57DA5D8 and EAID_dstE70932_F57C_4861_BE52_91F3F7270840 respectively. These elements are not contained in the XMI.~~ edit: targets are contained but not exported

**Comment by gkostkowski** [comment]
The issue has been resolved and the change has been included in the [2.3.0-rc.1](https://github.com/OP-TED/model2owl/releases/tag/2.3.0-rc.1) release. 
The paragraphs below provide explanation of the problem and the implemented solution.

**Problem**
In the former implementation of Model2Owl, the tool checked the prefix of a concept when the `generateReusedConcepts` filter was set to false. However, this logic was applied in some parts of the system even without being related to the filter. If the prefix differed from the `base-ontology-uri` variable (e.g `epo`), the concept was excluded from the generated artifacts. Since the ePO ontology uses modules like `eOrdering` and `eNotice` with prefixes such as `epo-not` and `epo-ord`, these were mistakenly considered external and were not generated.

**Solution**
New configuration variables are introduced to control artifact generation more precisely (`generateReusedConceptsSHACL`, `generateReusedConceptsOWLcore`, `generateReusedConceptsOWLrestrictions`, `generateReusedConceptsGlossary`). These variables allows individual control over whether reused concepts (considered external) are generated for each artifact type. 
Any parts of the system that previously applied the external logic without being linked to the `generateReusedConcepts` filter has been modified modified to connect with the new filters, ensuring consistent behavior. 

Additionally `includedPrefixesList` variable has been introduced. It contains a list of prefixes (e.g., `epo`, `epo-not`, `epo-ord`). These prefixes will always be treated as internal, ensuring that concepts with these prefixes are generated regardless of their designation as "reused."
